src/AppInstallerRepositoryCore/Rest/Schema/1_0/Interface.cpp

```diff
@@ -0,0 +1,143 @@
+            utility::string_t versionApi = utility::conversions::to_string_t(versionEndpoint);
+            return versionApi;
+        }
+
+        std::string GetStringFromJsonStringValue(const json::value& value)
+        {
+            return value.is_null() ? "" : utility::conversions::to_utf8string(value.as_string());
+        }
+    }
+
+    IRestClient::SearchResult Interface::Search(const std::string& restApiUri, const SearchRequest& request) const
+    {
+        UNREFERENCED_PARAMETER(request);
+        SearchResult result;
+
+        // TODO: Handle continuation token.
+        HttpClientHelper clientHelper(GetSearchEndpoint(restApiUri));
+        json::value jsonObject = clientHelper.HandlePost(GetSearchBody(request));
+
+        // Parse json and add results to SearchResult.
+        if (jsonObject.is_null())
+        {
+            return result;
+        }
+
+        auto& dataArray = jsonObject.at(Data).as_array();
+
+        for (auto& manifestItem : dataArray)
+        {
+            std::string packageId = GetStringFromJsonStringValue(manifestItem.at(PackageIdentifier));
+            std::string packageName = GetStringFromJsonStringValue(manifestItem.at(PackageName));
+            std::string publisher = GetStringFromJsonStringValue(manifestItem.at(Publisher));
+            std::string packageFamilyName = GetStringFromJsonStringValue(manifestItem.at(PackageFamilyName));
+            std::string productCode = GetStringFromJsonStringValue(manifestItem.at(ProductCode));
+            json::value versionValue = manifestItem.at(Versions);
+
+            if (packageId.empty() || packageName.empty() || publisher.empty() || versionValue.is_null() || versionValue.as_array().size() == 0)
+            {
+                continue;
+            }
+
+            std::vector<VersionAndChannel> versionList;
+            for (auto& versionItem : versionValue.as_array())
+            {
+                std::string version = GetStringFromJsonStringValue(versionItem.at(Version));
+                std::string channel = GetStringFromJsonStringValue(versionItem.at(Channel));
+                versionList.emplace_back(VersionAndChannel(version, channel));
+            }
+
+            PackageInfo packageInfo = PackageInfo(packageId, packageName, publisher);
+            Package package = Package(std::move(packageInfo), std::move(versionList));
+            result.Matches.emplace_back(std::move(package));
+        }
+
+        return result;
+    }
+
+    std::optional<std::string> Interface::GetManifestByVersion(const std::string& restApiUri, const std::string& packageId, const std::string& version) const
+    {
+        HttpClientHelper clientHelper(GetManifestByVersionEndpoint(restApiUri, packageId, version));
+        json::value jsonObject = clientHelper.HandleGet();
+
+        if (jsonObject.is_null())
+        {
+            return std::string();

```

Comment:
In this case you would want to `return {};` to return an empty optional, and probably want to preserve that when you change the function to return `std::optional<Manifest>`.